src/python/grpcio/grpc/__init__.py

```diff
@@ -753,6 +753,41 @@ def peer(self):
         raise NotImplementedError()
 
     @abc.abstractmethod
+    def peer_identity(self):
+        """Gets the peer identity(s).
+
+      Equivalent to ctx.auth_context().get(ctx.peer_identity_key())
+
+    Returns:
+      An iterable of the identities, or None if the call is not authenticated.
+      Each identity is returned as a raw bytes type.
+     """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def peer_identity_key(self):
+        """The auth property used to identify the peer.
+
+    For example, "x509_common_name" or "x509_subject_alternative_name" are
+    used to identify an SSL peer.
+    
+    Returns:
+      The auth property (string) that indicates the
+      peer identity, or None if the call is not authenticated.
+    """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def auth_context(self):
+        """Gets the auth context for the call.
+
+      Returns:
+        A multi-map of strings to iterables for each auth property.
```

Comment:
multi-map isn't an established, defined type in Python. Is this just a dictionary from strings to iterables of bytes?